s3/compress.go

```diff
@@ -230,7 +231,7 @@ func (c *Compressor) Upload(ctx context.Context, body io.Reader, bucket, key *st
 	return uploader.Upload(ctx, input)
 }
 
-func (c *Compressor) CreateObject(ctx context.Context, db models.DBInterface, body io.Reader, urlparams *models.URLParams, payload *models.ExportPayload) error {
+func (c *Compressor) CreateObject(ctx context.Context, db models.DBInterface, body io.Reader, urlparams *middleware.URLParams, payload *models.ExportPayload) error {
```

Comment:
It looks like this method really just needs the `ResourceUUID` instead of all the values in the struct.  If we modify this method to accept just the `ResourceUUID` instead of the entire `URLParams` struct then we can remove this dependency.  This way the `s3` package does not know about or depend on the `middleware` package.